chain/network/src/peer_manager/peer_store.rs

```diff
@@ -280,12 +300,10 @@ impl PeerStore {
                 to_remove.push(peer_id.clone());
             }
         }
-        let mut store_update = self.store.store_update();
-        for peer_id in to_remove {
-            self.peer_states.remove(&peer_id);
-            store_update.delete(DBCol::Peers, &peer_id.try_to_vec()?);
+        for peer_id in to_remove.iter() {
+            self.peer_states.remove(peer_id);
```

Comment:
Maybe move this to `delete_from_db` as well? It would be a noop when called in `new` and it feels like it belongs there so that the function makes sure the node is deleted from the hash as well as from the database.

Reply:
I think there are two ways to implement this:

1. Add parameter `&mut self` to `delete_from_db` and then in `PeerStore::new()`:
```
let peer_store = PeerStore { ... };
peer_store.delete_from_db(...)?; // method call after the instance was created
Ok(peer_store)
```
2. Instead of `&mut self`, pass the `peer_states` map as additional parameter to    `delete_from_db`.

Since I'm not sure if either way is optimal, I thought I'd bring it up before changing that part.

Reply:
Indirectly related as it might affect the refactor: `PeerStore::remove_expired()` removes peers from `peer_states` but not from the reverse `addr_peers`. Shouldn't they be removed from `addr_peers` as well?

Reply:
Passing `&mut self` sounds good to me.

As for removing from add_peers, I honestly don’t know.

Reply:
Function `delete_from_db` is updated and renamed to `delete_peers`, since it's more general now.

According to my understanding, the peer should be removed from `addr_peers` as well. Even if that's correct, however, maybe it's addressed rather in a separate PR.